### ISSUE TITLE
Pan left/right doesn't effect extension blocks

### ISSUE THREAD

**Issue by 2br-2b** [LLK/scratch-vm#2526]
### Expected Behavior

[When green flag clicked
set [pan left/right v] effect to (100)
play note (60) for (0.25) beats
speak (hello)](https://scratchblocks.github.io/#?style=scratch3&script=When%20green%20flag%20clicked%0Aset%20%5Bpan%20left%2Fright%20v%5D%20effect%20to%20(100)%0Aplay%20note%20(60)%20for%20(0.25)%20beats%0Aspeak%20(hello))

(I would post a Scratch blocks image, but Github isn't letting me upload the image)

All of the sound should play through the right speaker.

### Actual Behavior

The sound plays equally through both speakers.

The pan left/right causes surround sound, and should move all sound played to only the left/right speaker at 100%. However, this only works with the Sounds blocks, and not with other sound-playing blocks.

### Steps to Reproduce

Build the listed code in Scratch,

### Operating System and Browser

Linux Mint, Firefox

**Comment by ericrosenbaum** [comment]
We initially decided not to do this (i.e. not to make the pan effect work for sounds from extensions) for a design reason: we wanted the extensions to be functionally separate from the core blocks. It seemed potentially confusing to have something that a core block does (such as setting the pan value) alter something that an extension does (such as playing a musical note).

From the start, there was an exception to this separation, because in earlier versions of Scratch the "set volume" block affects the volume of "play note" and "play drum", which are now in the music extension. But we decided to keep the separation everywhere else. That's why the pitch and pan effects don't affect the music extension, and why volume, pitch and pan don't affect the text to speech extension. 

There are also technical reasons why the separation made sense- the architecture of the audio engine, and the way extensions that play sound make use of it, would have to change, especially in order to use the pitch effect on extension sounds.

In retrospect, I can see that people probably expect these features to work together. It also would be fun and expressive to be able to e.g. alter the pitch and pan of text to speech voices, or apply a pitch bend to a musical note. Unfortunately, I think that because people have been making projects that rely on the system working the way it does now, we can't change it, because we might break existing projects (for example, if you use the volume effect for play sound blocks, and have it set to zero while a character is talking with text to speech, that sound would now be muted as well). 

I'm reconsidering now, though. Previously I had assumed we needed to try to make a single consistent system level design choice (either core sound blocks should interact with extensions, or not), even though there's already an extension to that rule (with volume and the music extension). But maybe we could consider a smaller change, such as enabling just the pan effect to work on music and text to speech. This might change the behavior of existing projects unexpectedly, but not really break them... I'll raise this question with our design team.

**Comment by 2br-2b** [comment]
Yeah, I'm not sure that just applying the effects to the extensions is necessarily the best idea. I would like the effects to be applied to extensions, but just making it apply now would probably break some animations.

Maybe we could make separate effects blocks for the effects for the extensions? I also don't think this is the best idea, as it would probably overcomplicate things, but maybe have `change [extension v] [pitch v] effect by (10)` or `change text to speech [pitch v] effect by (100)` blocks? In other words, make blocks specifically to change the effects on the extensions.

Another option is to add an "add Text to Speech" option in the sounds tab where you can choose to record your voice or upload another sound file. That way, the text to speech saved as a sound and can be manipulated as such.
